docs/pages/2020_BIGMAP/sections/basics.rst

```diff
@@ -3,3 +3,658 @@
+
+Using the builder, you can easily set up the calculation by directly providing the input arguments.
+Let's use the ``Int`` node that was created by our previous ``calcfunction`` as one of the inputs and a new node as the second input:
+
+.. code-block:: ipython
+
+    In [3]: builder.x = load_node(pk=4)
+       ...: builder.y = Int(5)
+
+In case that your nodes' PKs are different and you don't remember the PK of the output node from the previous calculation, check the provenance graph you generated earlier and use the UUID of the output node instead:
+
+.. code-block:: ipython
+
+    In [3]: builder.x = load_node(uuid='42541d38')
+       ...: builder.y = Int(5)
+
+Note that you don't have to provide the entire UUID to load the node.
+As long as the first part of the UUID is unique within your database, AiiDA will find the node you are looking for.
+
+.. note::
+
+    One nifty feature of the builder is the ability to use tab completion for the inputs.
+    Try it out by typing ``builder.`` + ``<TAB>`` in the verdi shell.
+
+To execute the ``CalcJob``, we use the ``run`` function provided by the AiiDA engine:
+
+.. code-block:: ipython
+
+    In [4]: from aiida.engine import run
+       ...: run(builder)
+
+Wait for the process to complete.
+Once it is done, it will return a dictionary with the output nodes:
+
+.. code-block:: ipython
+
+    Out[4]:
+    {'sum': <Int: uuid: 7d5d781e-8f17-498a-b3d5-dbbd3488b935 (pk: 8) value: 11>,
+    'remote_folder': <RemoteData: uuid: 888d654a-65fb-4da0-b3bc-d63f0374f274 (pk: 9)>,
+    'retrieved': <FolderData: uuid: 4733aa78-2e2f-4aeb-8e09-c5cfb58553db (pk: 10s)>}
+
+Besides the sum of the two ``Int`` nodes, the calculation function also returns two other outputs: one of type ``RemoteData`` and one of type ``FolderData``.
+See the :ref:`topics section on calculation jobs <topics:calculations:usage:calcfunctions>` for more details.
+Now, exit the IPython shell and once more check for *all* processes:
+
+.. code-block:: console
+
+    $ verdi process list -a
+    PK  Created    Process label             Process State    Process status
+    ----  ---------  ------------------------  ---------------  ----------------
+    185  39m ago    multiply                  ⏹ Finished [0]
+    189  1m ago     ArithmeticAddCalculation  ⏹ Finished [0]
+
+    Total results: 2
+
+    Info: last time an entry changed state: 59s ago (at 00:31:33 on 2020-11-24)
+
+We now see two processes in the list.
+One is the ``multiply`` calcfunction you ran earlier, the second is the ``ArithmeticAddCalculation`` CalcJob that you have just run.
+Grab the PK of the ``ArithmeticAddCalculation``, and generate the provenance graph.
+The result should look like the graph shown in :numref:`BIGMAP_2020_Basics_fig_calcjob_graph`.
+
+.. code-block:: console
+
+    $ verdi node graph generate 7
+
+.. _BIGMAP_2020_Basics_fig_calcjob_graph:
+.. figure:: include/images/calcjob_graph.png
+    :scale: 35
+    :align: center
```

Comment:
This calcjob was listed with pk 189 just above (see explanation in other comment).

On the other hand: this provenance shows the new calcjob and the previous calcfunction, but when I generated it like this I only got the calcjob. We need to update the graph and/or the command.

Reply:
> This calcjob was listed with pk 189 just above (see explanation in other comment).

Same as above, still have to fix these.

> On the other hand: this provenance shows the new calcjob and the previous calcfunction, but when I generated it like this I only got the calcjob. We need to update the graph and/or the command.

Ah, good catch. I'll be updating the graphs as well (since else they don't match the PK's anymore). However, I do want it to show both the calculation function and job.

Reply:
Mmm I think `verdi node graph generate -a N` should work.

Reply:
Actually, are you sure you used the right node as input for the `ArithmeticAddCalculation` (i.e. the output node from the `multiply` calcfunction)? If you use one of the input nodes it won't generate the desired graph.

Reply:
Ah, ok, that could also work I think. That is not [what the text says](https://github.com/aiidateam/aiida-tutorials/pull/278/files/60ab18ac8ff1c37072c3335c4755c1941f9d4d92#diff-4619dfe538ceb00f99bc55b5fa9d0763929f7a6644c0eb019c268c0782865201R418) thoug, right?

```
Grab the PK of the ``ArithmeticAddCalculation``, and generate the provenance graph.
```

Reply:
Ahh, I see what you mean. I will double check this, sorry for the false alarm.